src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -475,40 +442,6 @@ private static ModuleMetadata ReadModuleMetadataFrom(ObjectReader reader, Serial
         }
     }
 
-    private static ModuleMetadata GetMetadata(Stream stream, long length)
-    {
-        if (stream is UnmanagedMemoryStream unmanagedStream)
```

Comment:
the old code had to manage getting an UnmanagedMemoryStream  back or not.  But the new code knows it *always* gets one of those back (it is strongly typed to do so if it uses the real service).  So thsi method just became the top block, with the rest being dead.  it was then inlined to the callsite.